### ISSUE TITLE
wrong identation in boolean condition but no error shown

### ISSUE THREAD

**Issue by ericqwerty0** [eslint/eslint#11826]
**Tell us about your environment**

* **ESLint Version:** 5.16.0
* **Node Version:** 10.15.3
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
parser: babel-eslint@10.0.1
plugin: eslint-plugin-react@7.13.0

**Please show your full configuration:**

I use vscode-eslint and i found that no error is shown when the identation is not correct in boolean conditions. I also installed eslint globally and run the command 
```sh
eslint <file>
```
but still got no errors.

<details>
<summary>Configuration</summary>

```js
{
    "env": {
        "browser": true,
        "es6": true,
        "node": true,
		"mocha": true
    },
    "extends": [
        "eslint:recommended",
        "plugin:react/recommended"
    ],
    "parserOptions": {
        "ecmaFeatures": {
            "ecmaVersion": 2018,
            "jsx": true
        },
        "sourceType": "module"
    },
    "parser": "babel-eslint",
    "globals": {
		"NODE_ENV" : false,
		"REACT_APP_URL" : false,
        "REACT_APP_CLIENT" : false,
        "BUILD_ENV": false
	},
    "plugins": [
        "react"
    ],
    "settings": {
        "react": {
          "version": "detect"
        }
    },
    "rules": {
        "indent": [
            "error",
            "tab"
        ],
        "no-global-assign": "off",
        "react/prop-types": [0]
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
if (((typeof r[0] === "string" || r[0] instanceof String) && r[0] === "") &&
														((typeof r[1] === "string" || r[1] instanceof String) && r[1] === "") &&
		((typeof r[2] === "string" || r[2] instanceof String) && r[2] === "") &&
									((typeof r[3] === "string" || r[3] instanceof String) && r[3] === "")) 
								return;
```

run in vscode

**What did you expect to happen?**
"Expected indentation of x tabs but found x." on the line with boolean conditions after "&&".
The error is shown on the line with "return;".

**What actually happened? Please include the actual, raw output from ESLint.**
No error

**Comment by kaicataldo** [comment]
Sorry for the delay in response. I have two questions:

1. Does this issue also occur with the default parser?
1. It's not entirely clear to me what the behavior you're expecting is. Can you be more specific with the actual number of tabs you're expecting and exactly on which line? It would also help us a lot if you can simplify this and find the minimum reproduction case.
